Kernel/Arch/x86/Processor.h

```diff
@@ -396,6 +402,22 @@ class Processor {
         return m_features.has_flag(feature);
     }
 
+    ALWAYS_INLINE static InterruptsState interrupts_state()
```

Comment:
This code dupe is unfortunate ... I wonder if we could have a Processor class that has generic functionality in it, and that can inherit from the arch specific implementation?

That way, we avoid any potential need for virtual inheritance?

Feel free to just slap a `FIXME`, on there for now.

Reply:
I created a new file `Arch/Processor.cpp` and put the common code in there. Eventually this file should move out of the `Arch` folder, since there is nothing architecture specific about it, but I kept it in the same directory as `Arch/Processor.h` for now.